tex/generic/pgf/systemlayer/pgfsys-dvips.def

```diff
@@ -159,28 +164,491 @@
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@color burn\endcsname{ColorBurn}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@hard light\endcsname{HardLight}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@soft light\endcsname{SoftLight}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@difference\endcsname{Difference}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@exclusion\endcsname{Exclusion}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@saturation\endcsname{Saturation}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@color\endcsname{Color}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@hue\endcsname{Hue}
+\expandafter\def\csname pgf@sys@pdf@blend@mode@map@luminosity\endcsname{Luminosity}
+
+
+%
+% Shadings
+%
+
+\def\pgfsys@horishading#1#2#3{%
+  {%
+    \pgf@parsefunc{#3}%
+    \pgf@xb=\pgf@max
+    \pgfmathsetlength\pgf@y{#2}%
+    \pgf@yb=\pgf@y%
+    \pgf@sys@bp@correct{\pgf@xb}%
+    \pgf@sys@bp@correct{\pgf@yb}%
+    \xdef\pgfutil@tempa{\hbox to \the\pgf@max{%
+      \noexpand\vrule width0pt height\the\pgf@y%
+      \noexpand\pgfsys@beginpurepicture%
+        \noexpand\pgfsys@outerinvoke{
+          \pgf@sys@tonumber{\pgf@yb} \pgf@sys@tonumber{\pgf@xb} 0 0 pgfe <<
+            /PatternType 2
+            /Shading <<
+              /ShadingType 2
+              /ColorSpace \pgf@shading@device\space
+              /Domain [\pgf@pdfparseddomain]
+              /Coords [\pgf@doma\space0 \pgf@domb\space0]
+              /Function \pgf@pdfparsedfunction
+            >>
+          >> matrix makepattern setpattern fill
+        }%
+        \hss%
+      \noexpand\pgfsys@endpurepicture%
+    }}%
+  }%
+  \global\expandafter\let\csname @pgfshading#1!\endcsname=\pgfutil@tempa%
+}
+\def\pgfsys@vertshading#1#2#3{%
+  {%
+    \pgf@parsefunc{#3}%
+    \pgfmathsetlength\pgf@x{#2}%
+    \pgf@xb=\pgf@x%
+    \pgf@yb=\pgf@max
+    \pgf@sys@bp@correct{\pgf@xb}%
+    \pgf@sys@bp@correct{\pgf@yb}%
+    \xdef\pgfutil@tempa{\hbox to \the\pgf@x{%
+      \noexpand\vrule width0pt height\the\pgf@max%
+      \noexpand\pgfsys@beginpurepicture%
+        \noexpand\pgfsys@outerinvoke{
+          \pgf@sys@tonumber{\pgf@yb} \pgf@sys@tonumber{\pgf@xb} 0 0 pgfe <<
+            /PatternType 2
+            /Shading <<
+              /ShadingType 2
+              /ColorSpace \pgf@shading@device\space
+              /Domain [\pgf@pdfparseddomain]
+              /Coords [0 \pgf@doma\space0 \pgf@domb]
+              /Function \pgf@pdfparsedfunction
+            >>
+          >> matrix makepattern setpattern fill
+        }%
+        \hss%
+      \noexpand\pgfsys@endpurepicture%
+    }}%
+  }%
+  \global\expandafter\let\csname @pgfshading#1!\endcsname=\pgfutil@tempa%
+}
+\def\pgfsys@radialshading#1#2#3{%
+  {%
+    \pgf@parsefunc{#3}%
+    \pgf@process{#2}%
+    \pgf@xa=\pgf@x%
+    \pgf@ya=\pgf@y%
+    \pgf@process{\pgfpoint{\pgf@max}{\pgf@max}}%
+    \advance\pgf@xa by \pgf@x%
+    \advance\pgf@ya by \pgf@y%
```

Comment:
Same for arithmetic, no `%`
```suggestion
    \advance\pgf@ya by \pgf@y
```

Reply:
Again, many occurrences of terminating `%` at other locations using `\advance`, in other files. Do you plan to fix them all?

Reply:
No, I won't fix any of them, but I don't want to have it in new code.